src/audio-worklet/basic/noise-generator/main.js

```diff
@@ -20,15 +22,28 @@ const startAudio = async (context) => {
   modulator.start();
 };
 
-// A simplem onLoad handler. It also handles user gesture to unlock the audio
+const stopAudio = () => {
+  if (noiseGenerator) {
+    noiseGenerator.disconnect();
+    isPlaying = false;
+  }
+};
+
+// A simple onLoad handler. It also handles user gesture to unlock the audio
 // playback.
 window.addEventListener('load', async () => {
   const buttonEl = document.getElementById('button-start');
   buttonEl.disabled = false;
+
   buttonEl.addEventListener('click', async () => {
-    await startAudio(audioContext);
-    audioContext.resume();
-    buttonEl.disabled = true;
-    buttonEl.textContent = 'Playing...';
-  }, false);
+    if (!isPlaying) { // If not playing, start audio
```

Comment:
nit: Google's JS style doesn't use this type of inline comments.

Reply:
moved the comments to next line.